src/quickunlock/Polkit.cpp

```diff
@@ -82,18 +83,37 @@ bool Polkit::getKey(const QString& dbPath, QByteArray& key)
     if (!hasKey(dbPath))
         return false;
 
+    pid_t app_pid = QCoreApplication::applicationPid();
+
+    QString processStatPath = QString("/proc/%1/stat").arg(app_pid);
+    QFile processStatFile(processStatPath);
+
+    if(!processStatFile.open(QIODevice::ReadOnly | QIODevice::Text)){
+        qDebug() << "polkit: failed to open " << processStatPath << " for reading";
+        return false;
+    }
+
+    QTextStream processStatStream(&processStatFile);
+    QString processStatInfo = processStatStream.readLine();
+    processStatFile.close();
+
+    QString processStartTime = processStatInfo.split(" ")[21];
+
+
+
     PolkitSubject subject;
     subject.kind = "unix-process";
-    subject.details.insert("pid", static_cast<uint>(QCoreApplication::applicationPid()));
-    subject.details.insert("start-time", static_cast<qulonglong>(0));
+    subject.details.insert("pid", static_cast<uint>(app_pid));
+    subject.details.insert("start-time", processStartTime.toULongLong());
 
     QMap<QString, QString> details;
-
+    
     auto result = m_polkit->CheckAuthorization(subject,
-                                               // "org.keepassxc.KeePassXC.unlockDatabase",
-                                               "org.freedesktop.policykit.exec",
+                                               "org.keepassxc.KeePassXC.unlockDatabase",
                                                details,
-                                               0x00000001,
+                                               0x00000001, 
+                                               // AllowUserInteraction - wait for user to authenticate
```

Comment:
If there's a usable header for this, it'd be more readable to just use the enum constant. But this is ok too.

Reply:
There are no header libraries available